srv/modules/runners/changed.py

```diff
@@ -74,6 +78,31 @@ def dependencies(self):
         """
         return self._depends
 
+    @property
+    def depends(self):
+        """
+        Getter for name attr
+        """
+        return self._depends
+
+    @depends.setter
+    def depends(self, role):
+        """
+        Setter for name attr
+        """
+        self._depends = role
+
+    def _set_depends(self):
+        """ Set dependency to self if not global """
+        if self._role_name == 'global':
+            # if the global config changed
+            # stage restart on the core services.
+            self.add_dependencies(Role(role_name='mon'))
+            self.add_dependencies(Role(role_name='mgr'))
+            self.add_dependencies(Role(role_name='storage'))
```

Comment:
`mds` too?

Reply:
I limited the restarts to the core services, since the mds should have it's own section (mds.conf)

Reply:
While that's true, I don't know if any of us know of all the Ceph variables that could impact both mds and another component.  If an admin updates the global.conf not quite understanding that it's more suited for mds.conf, I would still expect DeepSea to restart everything.  Currently, the mon, mgr and osds will restart, but not the mds.  So, the admin will still have suffer the penalty of the serial restart, but still have to investigate what went wrong.

Reply:
I get the point, but do we also stage 'rgws' for restarts then?

Reply:
I think so, which then gets us into `rgw_configurations`.  sigh... Considering that's not a one-liner, should that just be a separate PR.  I don't know the urgency of this patch currently.

Reply:
Yeah I think the issue is that _any_ config option can potentially be in the global section. Techincally we'd have to restart the whole cluster on a change here.

This is certainly very heavy-handed. Not sure what the reasonable way is.

Reply:
It is, but I swear that's where we were with this originally.  Since Ceph literally has 1K+ parameters (and maybe users shouldn't use them, but that doesn't stop them), we have no map to say `global X` impacts components A and B.  And even if we did, our maintained version would likely be wrong the minute we shipped.

Restarting the whole cluster gracefully is expensive, but anything less results in a bug like this one.

Reply:
Changed it to also restart `rgw` and `mds` - `igw` is not maintained in that runner. However I'd argue this is the right interface to stage restarts.